src/data/polynomial/degree/reverse_mul.lean

```diff
@@ -0,0 +1,605 @@
+  f.nat_degree = f.support.max' (non_zero_iff.mpr h) :=
+begin
+  apply le_antisymm,
+  { apply finset.le_max',
+    rw mem_support_iff_coeff_ne_zero,
+    exact leading_coeff_nonzero_of_nonzero.mp h, },
+  { apply max'_le,
+    intros y hy,
+    exact le_degree_of_mem_supp y hy, }
+end
+
+lemma support_term_nonzero {c : R} {n : ℕ} (h : c ≠ 0): (C c * X ^ n).support = singleton n :=
+begin
+  ext1,
+  rw mem_singleton,
+  split,
+    { exact mem_support_term, },
+    { intro,
+      rwa [mem_support_iff_coeff_ne_zero, ne.def, a_1, coeff_C_mul, coeff_X_pow_self n, mul_one], },
+end
+
+@[simp] lemma nat_degree_monomial {a : R} (n : ℕ) (ha : a ≠ 0) : nat_degree (C a * X ^ n) = n :=
+begin
+  rw defs_by_Johann,
+    { simp_rw support_term_nonzero ha,
+      exact max'_singleton n, },
+    { intro,
+      apply ha,
+      rw [← C_inj, C_0],
+      apply mul_X_pow_eq_zero a_1, },
+end
+
+lemma nat_degree_term_le (a : R) (n : ℕ) : nat_degree (C a * X ^ n) ≤ n :=
+begin
+  by_cases a0 : a = 0,
+    rw [a0, C_0, zero_mul, nat_degree_zero],
+    exact nat.zero_le _,
+
+
+    rw defs_by_Johann,
+      { simp_rw [support_term_nonzero a0, max'_singleton n], },
+      { intro,
+        apply a0,
+        rw [← C_inj, C_0],
+        apply mul_X_pow_eq_zero a_1, },
+end
+
+
+lemma nat_degree_term_nonzero {a : R} (n : ℕ) (ha : a ≠ 0) : nat_degree (C a * X ^ n) = n :=
+begin
+  rw defs_by_Johann,
+    { simp_rw [support_term_nonzero ha, max'_singleton n], },
+    { intro,
+      apply ha,
+      rw [← C_inj, C_0],
+      apply mul_X_pow_eq_zero a_1, },
+end
+
+
+@[simp] lemma trailing_coeff_eq_zero : trailing_coeff f = 0 ↔ f = 0 :=
+⟨λ h, by_contradiction $ λ hp, mt mem_support_iff.1
+  (not_not.2 h) (mem_of_min (trailing_degree_eq_nat_trailing_degree hp)),
+λ h, h.symm ▸ leading_coeff_zero⟩
+
+lemma trailing_coeff_nonzero_of_nonzero : f ≠ 0 ↔ trailing_coeff f ≠ 0 :=
+begin
+  apply not_congr trailing_coeff_eq_zero.symm,
+end
+
+
+lemma nat_trailing_degree_mem_support_of_nonzero : f ≠ 0 → nat_trailing_degree f ∈ f.support :=
+begin
+  rw mem_support_iff_coeff_ne_zero,
+  exact trailing_coeff_nonzero_of_nonzero.mp,
+end
+
+lemma nat_trailing_degree_le_of_mem_supp (a : ℕ) :
+  a ∈ f.support → nat_trailing_degree f ≤ a:=
+begin
+  rw mem_support_iff_coeff_ne_zero,
+  exact nat_trailing_degree_le_of_ne_zero,
+end
+
+lemma defs_by_Johann_trailing {R : Type*} [semiring R] {f : polynomial R} (h : f ≠ 0) :
+  nat_trailing_degree f = f.support.min' (non_zero_iff.mpr h) :=
+begin
+  apply le_antisymm,
+  { apply le_min',
+    intros y hy,
+    exact nat_trailing_degree_le_of_mem_supp y hy },
+  { apply finset.min'_le,
+    rw mem_support_iff_coeff_ne_zero,
+    exact trailing_coeff_nonzero_of_nonzero.mp h, },
+end
+
+/-- remove_leading_coefficient of a polynomial f is the polynomial obtained by
+subtracting from f the leading term of f. -/
+def remove_leading_coefficient (f : polynomial R) : polynomial R := begin
```

Comment:
We have some lemmas about this, in the form of `finsupp.erase`. I find it pretty useful, so I think it should have a definition that doesn't run into mild coercion problems you get from dealing with `finsupp`s. However, there are definitely easier ways to define this polynomial, like using `finsupp.erase`, or probably better, `f - C f.leading_coeff * X ^ f.nat_degree`, so that your docstring is more literally true.

Reply:
I did not know about the existence of ```erase```.  I thought of subtracting the leading coefficient, but that requires working with a ring, as opposed to a semiring, so I opted for getting rid of the support instead.